### DISCUSSION TITLE
Append dynamic views via dsl files after adding relations via Java code

### DISCUSSION THREAD

**Discussion by blancqua** [structurizr/java#363] [Q&A] [answered]
Hello,

We've been following the ([now removed](https://github.com/structurizr/examples/commit/78047a6fc99569e88ae08a104088a9e6c425f3f6#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R5)) example to model the software landscape of the organization. This means we have:
- one workspace DSL for the landscape, containing only personas and systems
- one workspace DSL for each system, extending from the landscape, including relationships
- relationships automatically added using Java code to the former by scanning the latter

Now I want to add a dynamic view on landscape level, using a DSL describing the view. The problem is as follows:
- I cannot just !include this dsl in the landscape dsl, since the relationships are not defined there, and that returns a (validation) error.
- Hence, I'd need to use the "augmented" workspace including the discovered relationships, but I do not find any elegant way to read a "partial DSL" containing the dynamic views for landscape level and inject it AFTER the relationships have been added dynamically.

So basically, it's starting from a DSL, adding some relationships dynamically via java, and then adding some more DSL to the workspace.

Let me know if you need more clarification but any help would be welcome.

**Comment by blancqua** [comment, marked as answer]
Good news, I managed to find a solution as follows:
- root.dsl, containing all systems, persons and styling, but no relations or lower-level (i.e. container) details
- for each system a DSL which extends from above
- 1 DSL for the landscape which extends from root.dsl, with a !plugin (instead of pure java code) augmenting the model with discovered relations and an !include later in the file to inject dynamic views that assume the relations are defined:

```
workspace extends ../root.dsl {

  !adrs adr
  !docs doc

  name "_Landscape_"

  !plugin <link to plugin which discovers and adds relations>

  views {
    !include dynamic.dsl
  }

  configuration {
    scope landscape
  }

}
```

Hope this helps if someone encounters an alike challenge.
